homeassistant/components/linky/sensor.py

```diff
@@ -49,22 +52,40 @@
     }
 )
 
+_LOGGER.error("LINKY_SENSOR")
+
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Configure the platform and add the Linky sensor."""
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-    timeout = config[CONF_TIMEOUT]
+    """Set up Linky sensor from legacy config file."""
+    _LOGGER.error("LINKY_SENSOR:setup_platform")
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
```

Comment:
We shouldn't set up the entry from the platform setup.

The platform schema should move to the component and be made a config schema.

Replace `def setup_platform` with `async def async_setup_platform` and just `pass` here.